aoc/src/main/scala/Day01.scala

```diff
@@ -8,22 +8,42 @@ object Day01 extends App:
   val start1: Long =
     System.currentTimeMillis
 
-  val commands: List[Char] =
+  val input: List[String] =
     Source
-      .fromResource(s"input$day.txt")
-      .mkString
-      .trim
-      .toList
+      .fromResource(s"input$day.txt").getLines.toList
+
+  case class Elves(elf: Int, amountOfCalories: Int)
+
+  def calculateCaloriesPerElf(calories: List[String], current: Elves, elves: List[Elves]): List[Elves] =
+    if calories.isEmpty then elves
+    else if calories.head.nonEmpty then
+      val newAmount = Elves(current.elf, current.amountOfCalories + calories.head.toInt)
+      calculateCaloriesPerElf(calories.tail, newAmount, elves :+ newAmount)
+    else if calories.head.isEmpty then
+      val newElf = Elves(current.elf + 1, 0)
+      calculateCaloriesPerElf(calories.tail, newElf, elves :+ newElf)
+    else sys.error("Unknown input")
+
+  val listElves: List[Elves] = calculateCaloriesPerElf(input, Elves(1,0), List())
```

Comment:
this works perfectly - and is very readable - but could be coded much more succinct using the library functions on `List` - so if you have time today.. try solving the puzzle after parsing the input into a `List[List[Int]]` in which each elf's calories represents one inner `List[Int]`

afterward the calculation is as easy as:

```val answer1 = input.map(_.sum).sorted.last```

and

```val answer2 = input.map(_.sum).sorted.takeRight(3).sum```

independent of that - your first golden star is earned - very nice :))